text-generation-inference/server/text_generation_server/jetstream_pt_support/logits_process.py

```diff
@@ -0,0 +1,80 @@
+from dataclasses import dataclass
+from typing import Tuple
+
+import jax
+import jax.numpy as jnp
+from transformers import GenerationConfig
+
+
+@dataclass
+class FusedLogitsWarper:
+    """
+    A class that performs top-k then top-p filtering, optionally applying a temperature.
+
+    Top-k filtering only keeps the `k` tokens with the best scores.
+
+    Top-p filtering only keeps the top tokens whose cumulated probability is above `p`.
+
+    The filtered tokens are returned as a list of indices, along with the corresponding subset of
+    the original logits.
+
+    If only top-k filtering is active, the filtered tokens are sorted by descending order.
+
+    If top-p filtering is active, the filtered tokens are sorted by ascending order.
+
+    Args:
+        temperature (`float`):
+            Strictly positive float value used to modulate the logits distribution. A value smaller than `1` decreases
+            randomness (and vice versa), with `0` being equivalent to shifting all probability mass to the most likely
+            token.
+        top_k (`int`):
+            The number of highest probability vocabulary tokens to keep for top-k-filtering.
+        top_p (`float`):
+            If set to < 1, only the smallest set of most probable tokens with probabilities that add up to `top_p` or
+            higher are kept for generation.
+    """
+
+    temperature: float = 1.0
+    top_k: int = 0
+    top_p: float = 1.0
+
+    @classmethod
+    def from_config(cls, generation_config: GenerationConfig) -> "FusedLogitsWarper":
+        """Instantiate a fused warper from a generation configuration.
+
+        Args:
+            generation_config (`~transformers.generation.GenerationConfig`):
+                The generation configuration to be used as base parametrization for the fused warper.
+
+        Returns:
+            a `FusedLogitsWarper` or None if neither top-k nor top-p are configured.
+        """
+        return cls(generation_config.temperature, generation_config.top_k, generation_config.top_p)
+
+    def __call__(self, logits: jnp.ndarray) -> Tuple[jnp.ndarray, jnp.ndarray]:
+        if self.temperature != 1.0:
+            logits = logits / self.temperature
```

Comment:
qq: what happens if temp = 0?

Reply:
Good question @miladm. In that the operation will give an array with [inf, -inf] values. The generation will still give some result, though probably not the one you would expect (in my case it was as if it was using greedy search).
BTW, you will have the same division in the Jetstream [sampling code](https://github.com/google/JetStream/blob/445f1aa8e857d0a09d72618e365daf80723bdf4c/jetstream/engine/sampling_utils.py#L40).